src/digraph-cyclicity.jl

```diff
@@ -0,0 +1,371 @@
+abstract Visitor
+
+"""
+```transitiveclosure!(dg::DiGraph)```
+
+Compute the transitive closure of a directed graph, using the Floyd-Warshall
+algorithm.
+"""
+function transitiveclosure!(dg::DiGraph)
+    for k in dg.vertices
+        for i in dg.vertices
+            for j in dg.vertices
+                if (has_edge(dg, i, k) & has_edge(dg, k, j))
+                    add_edge!(dg, i, j)
```

Comment:
Also: you're mutating state in the middle of a for loop. This is very problematic.